### PR TITLE
Dotnet cli tests

### LINKED ISSUE #67: `dotnet-cli` module cleanup
I stumbled upon the `dotnet-cli` module and it could use some cleanup based on newer patterns we have. Here's a quick list of what I suggest, ordered by priority:

- Check exit code from `shell.exec` and echo an error message if the code is not successful (0). This should kill the process chain with `shell.exit(result.code)` so that any commands leveraging it will not try to keep working on a failed CLI command.
- Consider refactoring from `shell.exec` to `child_process.spawnSync`, which streams output and retains original colors. 
    - If doing this refactor, make sure to update the `cmd()` function with the existing pattern in `dotnet-test`, `dotnet-build`, `dotnet-clean`, etc. It is a simple way to separate out the 'base' command from the arguments, as well as a `toString()` override for output purposes
- Reference the absolute path of the `Cli.dll` executable in `description()` to allow for easier debugging & visibility on where the executable is actually being executed from.
- Remove reference to LMS in the description
- Add test coverage

### PR DESCRIPTION
**Pull request by Drewcrew1**
Fixes #67 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

**Comment by Drewcrew1** [comment]
@brandongregoryscott I got the import added

File: _modules/dotnet-cli.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// -----------------------------------------------------------------------------------------
+// #region Imports
+// -----------------------------------------------------------------------------------------
+const child_process = require("child_process");
```
Comment (brandongregoryscott) [inline _modules/dotnet-cli.test.js:4]: Nitpick - add a new line between the start of a region and its code

File: _modules/dotnet-cli.test.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+const dotnetCli     = require("./dotnet-cli");
+const shell         = require("shelljs");
+const dotnetPath    = require("./dotnet-path");
+const dotnetBuild   = require("./dotnet-build");
+const faker         = require("faker");
+
+// #endregion Imports
+
+
```
Comment (brandongregoryscott) [inline _modules/dotnet-cli.test.js:13]: Nitpick - extra new line here

File: _modules/dotnet-cli.test.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// -----------------------------------------------------------------------------------------
+// #region Tests
+// -----------------------------------------------------------------------------------------
+
+describe("dotnetCli", () => {
+
+  // -----------------------------------------------------------------------------------------
+  // #region run
+  // -----------------------------------------------------------------------------------------
+  describe("run", () => {
```
Comment (brandongregoryscott) [inline _modules/dotnet-cli.test.js:23]: Nitpick - add a new line between the start of a region and its code

File: _modules/dotnet-cli.test.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+         // Act
+         dotnetCli.run();
+
+         // Assert
+         expect(dotnetPathSpy).toHaveBeenCalled();
+         expect(dotnetBuildSpy).toHaveBeenCalled();
+    });
+
+    //TODO: test if spawn sync returns a non 0 status
+    // it calls shell.exit with the status
```
Comment (brandongregoryscott) [inline _modules/dotnet-cli.test.js:43]: Are these comments still necessary? I think we covered them 👍

File: _modules/dotnet-cli.test.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      dotnetCli.run();
+
+      // Assert
+      expect(spawnSync).toHaveBeenCalled();
+      expect(shellExitSpy).toHaveBeenCalledWith(exitCode);
+    });
+
+  });
+
+
```
Comment (brandongregoryscott) [inline _modules/dotnet-cli.test.js:61]: Nitpick - extra new line here

**Pull request merged**